### PR TITLE
Use a sub request and a custom route for tokens

### PR DESCRIPTION
**Pull request by drsnyder**
Use a sub request and a custom route for token verification. This prevents the cosocket based http client object from being [shared across requests](http://wiki.nginx.org/HttpLuaModule#Data_Sharing_within_an_Nginx_Worker) and contexts which appears to cause problems.

It's odd that this doesn't appear to cause problems in production but does in dev and on OS X. I can't explain that but this approach appears to be more efficient and more idiomatic nginx+lua.

This also simplifies the networking in lua a little and allows us to get rid of the `net` module.

https://wikia-inc.atlassian.net/browse/SERVICES-638

Related: https://github.com/openresty/lua-nginx-module/issues/150

TODO:
 * [ ] update chef to pin `lua-resty-http` to 0.06-0

/cc @Wikia/services-team

### CODE REVIEW

File: nginx/production/api-gateway.conf [hunk 1]
```diff
@@ -31,6 +31,34 @@ server {
       ';
   }
 
+  location /token {
```
Comment (nmonterroso) [inline nginx/production/api-gateway.conf]: should this be `/auth/token`?

Comment (nmonterroso) [inline nginx/production/api-gateway.conf]: Nevermind, just realized what this is doing

**Comment by nmonterroso** [comment]
LGTM :shipit:

**Comment by harnash** [comment]
We got confirmation from one of the reporters that issue is resolved on dev. Merging.

**Pull request merged**